packages/test/test-end-to-end-tests/src/test/refreshSerializerLifeCycle.spec.ts

```diff
@@ -0,0 +1,195 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+
+import { generatePairwiseOptions } from "@fluid-private/test-pairwise-generator";
+import { describeCompat } from "@fluid-private/test-version-utils";
+import { Deferred } from "@fluidframework/core-utils/internal";
+import {
+	ITestFluidObject,
+	timeoutPromise,
+	DataObjectFactoryType,
+	createAndAttachContainer,
+	timeoutAwait,
+	waitForContainerConnection,
+	type ChannelFactoryRegistry,
+	type ITestObjectProvider,
+} from "@fluidframework/test-utils/internal";
+import type { IContainerExperimental } from "@fluidframework/container-loader/internal";
+import {
+	type IContainerRuntimeOptions,
+	DefaultSummaryConfiguration,
+} from "@fluidframework/container-runtime/internal";
+import type {
+	ConfigTypes,
+	IConfigProviderBase,
+	ITelemetryBaseLogger,
+} from "@fluidframework/core-interfaces/internal";
+import { SharedMap, type ISharedMap } from "@fluidframework/map/internal";
+import { MockLogger } from "@fluidframework/telemetry-utils/internal";
+import type { IHostLoader } from "@fluidframework/container-definitions/internal";
+import { wrapObjectAndOverride } from "../mocking.js";
+
+const testConfigs = generatePairwiseOptions({
+	savedOps: [true, false],
+	pendingOps: [true, false],
+	savedOps2: [true, false],
+	pendingOps2: [true, false],
+	summaryWhileOffline: [true, false],
+	waitForRefresh: [true,false],
+});
+
+describeCompat("Validate Attach lifecycle", "NoCompat", (getTestObjectProvider, apis) => {
```

Comment:
since you decided to build this as  compat tests, please ensure you validate that it works against all the server we test against